### PR TITLE
Test reorg

### PR DESCRIPTION
**Pull request by SaltyDavidselph**

### CODE REVIEW

**Review by eeeebbbbrrrr — CHANGES_REQUESTED** [review]
@SaltyDavidselph overall this looks good.  The only thing is I don't want this PR to change the `pgrx` version PL/Rust currently uses.  So please set them all back to `0.9.7`.

Once that's done and we confirm CI still works we can merge this.

File: plrust-tests/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+[features]
+default = ["pg13"]
+pg13 = ["pgrx/pg13", "pgrx-tests/pg13" ]
+pg14 = ["pgrx/pg14", "pgrx-tests/pg14" ]
+pg15 = ["pgrx/pg15", "pgrx-tests/pg15" ]
+pg_test = []
+trusted = []
+
+[dependencies]
+pgrx = "=0.9.8"
```
[the remark this answers is not in the document]
Comment (SaltyDavidselph) [reply plrust-tests/Cargo.toml]: Done

File: plrust-tests/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+tempdir = "0.3.7"
+once_cell = "1.18.0"
+
+[dev-dependencies]
+pgrx-tests = "=0.9.8"
```
[the remark this answers is not in the document]
Comment (SaltyDavidselph) [reply plrust-tests/Cargo.toml]: Done

File: plrust-trusted-pgrx/Cargo.toml [hunk 3]
```diff
@@ -18,7 +18,7 @@ pg15 = ["pgrx/pg15"]
 
 [dependencies]
 # changing the pgrx version will likely require at least a minor version bump to this create
-pgrx = { version = "=0.9.7", features = [ "no-schema-generation" ], default-features = false }
+pgrx = { version = "=0.9.8", features = [ "no-schema-generation" ], default-features = false }
```
[the remark this answers is not in the document]
Comment (SaltyDavidselph) [reply plrust-trusted-pgrx/Cargo.toml]: Done

File: plrust/Cargo.toml [hunk 4]
```diff
@@ -39,7 +39,7 @@ serde = "1.0.164"
 serde_json = "1.0.97"
 
 # pgrx core details
-pgrx = { version = "=0.9.7" }
+pgrx = { version = "=0.9.8" }
```
[the remark this answers is not in the document]
Comment (SaltyDavidselph) [reply plrust/Cargo.toml]: Done

File: plrust/Cargo.toml [hunk 5]
```diff
@@ -66,6 +66,6 @@ memfd = "0.6.3" # for anonymously writing/loading user function .so
 
 
 [dev-dependencies]
-pgrx-tests = { version = "=0.9.7" }
+pgrx-tests = { version = "=0.9.8" }
```
[the remark this answers is not in the document]
Comment (SaltyDavidselph) [reply plrust/Cargo.toml]: Done

File: Cargo.lock [hunk 6]
```diff
@@ -1686,6 +1749,16 @@ version = "1.0.1"
...
 
+[[package]]
+name = "tempdir"
+version = "0.3.7"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "15f2b5fb00ccdf689e0149d1b1b3c03fead81c2b37735d812fa8bddbbf41b6d8"
+dependencies = [
+ "rand 0.4.6",
+ "remove_dir_all",
+]
```
Comment (workingjubilee) [inline Cargo.lock:1760]: This indicates these test changes don't account for the diff in https://github.com/tcdi/plrust/pull/357

File: plrust-tests/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        let work_dir = TempDir::new("plrust-tests").expect("Couldn't create tempdir");
+        format!("plrust.work_dir='{}'", work_dir.path().display())
+    });
+    static LOG_LEVEL: &str = "plrust.tracing_level=trace";
+
+    static PLRUST_ALLOWED_DEPENDENCIES_FILE_NAME: &str = "allowed_deps.toml";
+    static PLRUST_ALLOWED_DEPENDENCIES_FILE_DIRECTORY: Lazy<TempDir> = Lazy::new(|| {
+        use std::io::Write;
+        let temp_allowed_deps_dir =
+            TempDir::new("plrust-allowed-deps").expect("Couldnt create tempdir");
```
Comment (workingjubilee) [inline plrust-tests/src/lib.rs:68]:
```suggestion
    use tempfile::{tempdir, TempDir};

    static WORK_DIR: Lazy<String> = Lazy::new(|| {
        let work_dir = tempdir().expect("Couldn't create tempdir");
        format!("plrust.work_dir='{}'", work_dir.path().display())
    });
    static LOG_LEVEL: &str = "plrust.tracing_level=trace";

    static PLRUST_ALLOWED_DEPENDENCIES_FILE_NAME: &str = "allowed_deps.toml";
    static PLRUST_ALLOWED_DEPENDENCIES_FILE_DIRECTORY: Lazy<TempDir> = Lazy::new(|| {
        use std::io::Write;
        let temp_allowed_deps_dir =
            tempdir().expect("Couldnt create tempdir");
```

File: plrust-tests/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+    pub fn postgresql_conf_options() -> Vec<&'static str> {
+        // return any postgresql.conf settings that are required for your tests
+        vec![
+            "shared_preload_libraries = 'plrust'",
+            "plrust.work_dir = '/tmp/plrust-work-dir'"
+        ]
+    }
+}
+*/
```
Comment (workingjubilee) [inline plrust-tests/src/lib.rs:147]: Pardon, but why is this commented out here?

File: plrust-tests/Cargo.toml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+default = ["pg13"]
+pg13 = ["pgrx/pg13", "pgrx-tests/pg13" ]
+pg14 = ["pgrx/pg14", "pgrx-tests/pg14" ]
+pg15 = ["pgrx/pg15", "pgrx-tests/pg15" ]
+pg_test = []
+trusted = []
+
+[dependencies]
+pgrx = "=0.9.7"
+tempdir = "0.3.7"
```
Comment (workingjubilee) [inline plrust-tests/Cargo.toml:19]:
```suggestion
tempfile = "0.3.8"
```

File: plrust-tests/Cargo.toml [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+once_cell = "1.18.0"
+
+[dev-dependencies]
+pgrx-tests = "=0.9.7"
+tempdir = "0.3.7"
```
Comment (workingjubilee) [inline plrust-tests/Cargo.toml:24]:
```suggestion
tempfile = "0.3.8"
```

File: plrust-tests/src/lib.rs (hunk 8 above)
Comment (SaltyDavidselph) [reply plrust-tests/src/lib.rs:147]: I believe it was a hold over from moving all the test here 
will be removed

File: plrust-tests/Cargo.toml (hunk 9 above)
Comment (workingjubilee) [reply plrust-tests/Cargo.toml]: Oh, this was the wrong suggestion. I'm silly.

**Comment by workingjubilee** [comment]
Rebased and pulled out the additional test added by tcdi/plrust#373.

**Comment by workingjubilee** [comment]
Somehow pulling out the test broke it, so I am going to leave that for a later PR to examine, this one has been waiting long enough.

**Pull request merged**